swigpyrun missing some MPI location variable

Which variable am I missing?

```
[ 97%] Generating swigpyrun.h
cd /tmp/trilinos-build/packages/PyTrilinos/src && /opt/apps/swig/3.0/bin/swig -python -external-runtime
cd /tmp/trilinos-build && /opt/apps/cmake/3.10.2/bin/cmake -E cmake_depends "Unix Makefiles" /admin/rpms/BUILD/trilinos-git /admin/rpms/BUILD/trilinos-git/packages/PyTrilinos/src /tmp/trilinos-bu
ild /tmp/trilinos-build/packages/PyTrilinos/src /tmp/trilinos-build/packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/DependInfo.cmake --color=
Traceback (most recent call last):
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 647, in <module>
    main()
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 546, in main
    print("#define MPI_VERSION %s" % get_mpi_version())
  File "/tmp/trilinos-build/packages/PyTrilinos/src/gen_teuchos_rcp.py", line 47, in get_mpi_version
    for line in open(header, 'r').readlines():
IOError: [Errno 2] No such file or directory: '/include/mpi.h'
make[2]: *** [packages/PyTrilinos/src/Teuchos.RCP.i] Error 1
make[2]: *** Deleting file `packages/PyTrilinos/src/Teuchos.RCP.i'
make[2]: Leaving directory `/tmp/trilinos-build'
make[1]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/all] Error 2
```

Is that a new variable? I used to specify the mpicc & such compilers.

Ok, with that variable:

```
/admin/rpms/BUILD/trilinos-git/packages/teuchos/comm/src/Teuchos_DefaultComm.hpp:291: Error: CPP #error ""Sorry, you need an MPI implementation that supports at least MPI 2.0 in order to build th
is code.  MPI 2.0 came out in 1997.  I wrote this comment in 2017.  If you really _really_ want MPI 1.x support, please file a GitHub issue for this feature request at github.com/trilinos/trilino
s/issues with an expression of its priority and we will get to it as soon as we can."". Use the -cpperraswarn option to continue swig processing.
/admin/rpms/BUILD/trilinos-git/packages/teuchos/comm/src/Teuchos_DefaultComm.hpp:385: Error: CPP #error ""Sorry, you need an MPI implementation that supports at least MPI 2.0 in order to build th
is code.  MPI 2.0 came out in 1997.  I wrote this comment in 2017.  If you really _really_ want MPI 1.x support, please file a GitHub issue for this feature request at github.com/trilinos/trilino
s/issues with an expression of its priority and we will get to it as soon as we can."". Use the -cpperraswarn option to continue swig processing.
make[2]: *** [packages/PyTrilinos/src/Teuchos.RCPPYTHON_wrap.cpp] Error 1
make[2]: Leaving directory `/tmp/trilinos-build'
make[1]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/all] Error 2
```

--- a/packages/PyTrilinos/src/gen_teuchos_rcp.py.in
+++ b/packages/PyTrilinos/src/gen_teuchos_rcp.py.in
@@ -46,7 +46,7 @@ def get_mpi_version():
     version = ""
     for line in open(header, 'r').readlines():
         if "MPI_VERSION" in line:
-            version = line.split(" ")[2]
+            version = line.split()[2]
             break
     return version
 
@@ -238,23 +238,12 @@ def form_class_filename(parsed_class):
     """
 
     # Determine the extension
-    if parsed_class["name"].startswith("Epetra_"):
-        ext = ".h"
-    elif parsed_class["name"].startswith("Amesos_"):
-        ext = ".h"
-    elif parsed_class["name"].startswith("AztecOO_"):
-        ext = ".h"
-    elif parsed_class["name"].startswith("Ifpack_"):
-        ext = ".h"
-    elif parsed_class["name"].startswith("Komplex_"):
+    if parsed_class["name"].startswith(("Epetra_", "Amesos_" , "AztecOO_",
+                                        "Ifpack_", "Komplex_", "Trilinos_Util_")):
         ext = ".h"
     elif parsed_class["name"] == "Pliris":
         ext = ".h"
-    elif parsed_class["name"].startswith("Trilinos_Util_"):
-        ext = ".h"
-    elif parsed_class["namespaces"][0] == "EpetraExt":
-        ext = ".h"
-    elif parsed_class["namespaces"][0] == "MLAPI":
+    elif parsed_class["namespaces"][0] in ("EpetraExt", "MLAPI"):
         ext = ".h"
     elif parsed_class["namespaces"][0] in ("NOX", "LOCA"):
         ext = ".H"
@@ -428,6 +417,9 @@ def reorder_classnames(classnames):
 ################################################################################
 
 def get_import_directives(teuchos_rcps):
+    """
+
+    """
     result = []
     for classname in teuchos_rcps:
         filename = form_class_filename(parse_class(classname))

--- a/packages/PyTrilinos/src/gen_teuchos_rcp.py.in
+++ b/packages/PyTrilinos/src/gen_teuchos_rcp.py.in
@@ -39,6 +39,13 @@ WITH_LOCA      = cmake_bool("${NOX_ENABLE_LOCA}"            )
 WITH_MPI       = cmake_bool("${TPL_ENABLE_MPI}"             )
 WITH_PETSC     = cmake_bool("${TPL_ENABLE_PETSC}"           )
 
+# Find the MPI base directory
+MPI_BASE_DIR   = "${MPI_BASE_DIR}"
+if WITH_MPI:
+    if MPI_BASE_DIR == "":
+        MPI_BIN_DIR  = os.path.split("${MPI_CXX_COMPILER}")[0]
+        MPI_BASE_DIR = os.path.split(MPI_BIN_DIR)[0]
+
 ################################################################################
 
 def get_mpi_version():

